### PR TITLE
Improve storage for local dev

### PR DESCRIPTION
**Pull request by tsibley**
Fixes for things I ran into during recent local dev sessions. See commit messages for details.

### Testing

- [x] Checks pass (though nothing here is used in CI)

### CODE REVIEW

File: devel/clean
Comment (victorlin) [inline devel/clean:42]:
```suggestion
echo "--> Deleting local images pulled from localhost:5000"
```

Clarify that this does not delete local images pulled from elsewhere (e.g. Docker Hub).

File: devel/start-localhost-registry
Comment (victorlin) [inline devel/start-localhost-registry:26]: _(non-blocking)_

Upon `make clean` I noticed that the new `build/` directory had a size of 2.1G. I tried to find the accumulated volumes from previous registry usage, but none had large sizes except for the builder volume which is used elsewhere. Building from `master` also does not produce any new volumes in `docker volume ls`.

I think I'm missing something here. Without the custom `--volume`, do the transient volumes get pruned somehow?

File: devel/clean
Comment (tsibley) [reply devel/clean:43]: Hmm. I think you're mistaken. It deletes what's shown in `docker image ls`, not the local registry. (The local registry's copies are deleted with `rm -rf "$build_dir"` earlier.)

`docker image pull localhost:5000/nextstrain/base:latest` results in `localhost:5000/nextstrain/base:latest` being available, e.g. in `docker image ls`. It doesn't result in `nextstrain/base:latest` being available.

File: devel/clean
Comment (tsibley) [reply devel/clean:44]: Fair enough. I skipped parameterization here because it seemed unnecessary, but I'll thread it thru.

(The overhead of threading thru params is why I prefer env vars, e.g. `LOCAL_REGISTRY=…`, over command-line arguments when there are many separate programs co-operating over shared resources.)

File: devel/clean
Comment (victorlin) [reply devel/clean:43]: Oops, I realized this later and thought I deleted this comment, but looks like that didn't happen. You're right!

File: devel/clean
Comment (victorlin) [reply devel/clean:44]: I'd be ok with an env var for the local registry, if you want to make that change here.

File: devel/start-localhost-registry
Comment (tsibley) [reply devel/start-localhost-registry:26]: Hmm. They don't get pruned for me, hence the reason for the change, and I can see them in `docker volume ls`. On `master`, for example:

    $ docker volume ls 
    DRIVER    VOLUME NAME

    $ ./devel/start-localhost-registry 

    $ docker volume ls 
    DRIVER    VOLUME NAME
    local     85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87

    $ docker container inspect nextstrain-local-registry -f '{{json .Mounts}}' | jq
    [
      {
        "Type": "volume",
        "Name": "85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87",
        "Source": "/var/lib/docker/volumes/85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87/_data",
        "Destination": "/var/lib/registry",
        "Driver": "local",
        "Mode": "",
        "RW": true,
        "Propagation": ""
      }
    ]

    $ ./devel/stop-localhost-registry 

    $ docker volume ls 
    DRIVER    VOLUME NAME
    local     85e2e9bfedc4a83eea0f00b978133d37b26fbd020294143b40f9c50d32291d87

    $ docker container inspect nextstrain-local-registry -f '{{json .Mounts}}' | jq
    Error: No such container: nextstrain-local-registry

Perhaps something is auto-pruning them for you, e.g. Docker Desktop or a newer Docker version?

Comment (tsibley) [reply devel/start-localhost-registry:26]: What does `docker system df` report for you?

File: devel/clean
Comment (tsibley) [reply devel/clean]: Done with repush in 23a8431.

File: devel/clean
Comment (tsibley) [reply devel/clean]: Parameterized with repush in 23a8431.

Comment (tsibley) [reply devel/clean]: I'll save any switch from command-line args to env vars for another time!

File: devel/start-localhost-registry
Comment (victorlin) [reply devel/start-localhost-registry:26]: Oh nevermind, I see the 2GB volume now. I think Docker Desktop's UI was displaying an incorrect size, or I misread earlier. 

I can reproduce the Docker Desktop UI to show these incorrect sizes after a change in "status":

image

Clicking out then back into the "Volumes" view shows numbers that better reflect what's reported by `docker system df -v`:

```
Local Volumes space usage:

VOLUME NAME                                                        LINKS     SIZE
3647c4b5c1767b95a75be5bcc79ede3ff2cce3e9b52759361f6b3cd3cc0cb96d   0         0B
81f2fbdfd9fa34492dfb49e6761ad50f845c380275cd4a5267276cdc23a39e85   0         2.13GB
buildx_buildkit_nextstrain-builder0_state                          1         5.456GB
d99f38bd772f04ead379fb7eadc20de6589e9c66dd64d3eb3ad605f2c36dd63f   0         0B
f974ecaf61113fe8596ba6bcceaadb364a7435e071a060383fac9e1934030097   0         0B
823334b73184b6465a727efd7835a373a0426650e734a9f4a7c5a85a5ac11f59   0         2.13GB
c0c7d34820a0ebb85914fcbdfe7d6981ff3877ee394866bf6476df506b18a4b1   1         0B
3002016aaf02647af6150de06e83ceec9544bfb8a207a94f801dfe5506764f81   0         0B
fc3b5d46e5200e3ae2ccf1302e1b2c2780a13a17f8cf80880eda6b3519dcc11f   0         0B
```

Comment (tsibley) [reply devel/start-localhost-registry:26]: :upside_down_face:

**Review by victorlin — APPROVED** [review]
I'm not sure what happened with CI, but changes look good.

**Comment by tsibley** [comment]
(CI failures were a GitHub Actions outage.)

**Comment by tsibley** [comment]
Rebased onto master, including a small fix caught by the new ShellCheck linting.

I'll wait for PR CI to pass before merging.

**Pull request merged**
